lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -20,11 +20,40 @@ import 'package:over_react_codemod/src/util.dart';
 typedef YieldPatch = void Function(
     int startingOffset, int endingOffset, String replacement);
 
-@visibleForTesting
-bool isPublicForTest = false;
+SemverHelper semverHelper;
 
-// Stub while <https://jira.atl.workiva.net/browse/CPLAT-9308> is in progress
-bool isPublic(ClassDeclaration node) => isPublicForTest;
+/// Returns whether or not [node] is publicly exported.
+bool isPublic(ClassDeclaration node) {
+  assert(semverHelper != null);
+  return semverHelper.getPublicExportLocations(node) != null;
+}
+
+class SemverHelper {
+  Map _exportList;
+
+  SemverHelper(Map jsonReport) {
+    _exportList = jsonReport['exports'];
+  }
+
+  /// Returns semver report information for [node] if it is publicly exported.
+  ///
+  /// If [node] is not publicly exported, returns `null`.
+  Map<String, dynamic> getPublicExportLocations(ClassDeclaration node) {
+    assert(_exportList != null);
```

Comment:
I would expect this to be done in the constructor, so that the error occurs early and reliably. FYI, you can also do that in the initializer list!
```dart
SemverHelper(Map jsonReport) : 
    _exportList = jsonReport['exports'],
    assert(_exportList != null);